src/components/ActionSheetAwareScrollView/ActionSheetKeyboardSpace.tsx

```diff
@@ -0,0 +1,411 @@
+            case States.ATTACHMENTS_POPOVER_WITH_KEYBOARD_OPEN: {
+                return interpolate(keyboard.progress.value, [0, 1], [popoverHeight - composerHeight, 0]);
+            }
+            case States.CALL_POPOVER_WITH_KEYBOARD_OPEN: {
+                if (keyboard.height.value > 0) {
+                    console.log("TRANSITION #14 (2) -> ", 0, { lastKeyboardHeight, popoverHeight, composerHeight }, new Date().getTime());
+
+                    return 0;
+                }
+
+                console.log("TRANSITION #14 (1) -> ", popoverHeight - composerHeight, { lastKeyboardHeight, popoverHeight, composerHeight }, new Date().getTime());
+
+                return setInitialValueAndRunAnimation(lastKeyboardHeight, withSpring(popoverHeight - composerHeight, config));
+            }
+            case States.CALL_POPOVER_WITH_KEYBOARD_CLOSED: {
+                // keyboard is opened
+                if (keyboard.height.value === keyboard.heightWhenOpened.value) {
+                    console.log("TRANSITION #14 (1-2) -> ", 0);
+                    return 0;
+                }
+                console.log("TRANSITION #14 (1-1) -> ", popoverHeight - composerHeight, { lastKeyboardHeight, popoverHeight, composerHeight }, new Date().getTime());
+                return withSpring(lastKeyboardHeight, config);
+            };
+            case States.EMOJI_PICKER_WITH_KEYBOARD_OPEN: {
+                if (keyboard.state.value === KeyboardState.CLOSED) {
+                    console.log("TRANSITION #14 -> ", popoverHeight - composerHeight, { lastKeyboardHeight, popoverHeight, composerHeight }, new Date().getTime());
+                    return popoverHeight - composerHeight;
+                    // return 451;
+                    // return withTiming(lastKeyboardHeight, {duration: 250});
+                    return withSequence(
+                        // artificial delay for one frame, because `paddingBottom` in KAV is updated only in next frame
+                        withTiming(0, {
+                            duration: 8,
+                        }),
+                        withTiming(lastKeyboardHeight, {
+                            duration: 0,
+                        }),
+                    );
+                    // return lastKeyboardHeight;
+                    /* console.log("TRANSITION #13 ->", 0);
+                    return 0; */
+                }
+
+                /* console.log("TRANSITION #14 -> ", lastKeyboardHeight);
+                return lastKeyboardHeight; */
+                console.log("TRANSITION #13 ->", 0);
+                return 0;
+            }
+
+            case States.KEYBOARD_POPOVER_CLOSED: {
+                // transition({type: Actions.END_TRANSITION});
+                if (keyboard.heightWhenOpened.value === keyboard.height.value) {
+                    console.log("TRANSITION #15 (1) -> ", 0, {keyboard: keyboard.height.value, lastKeyboardHeight, heightWhenOpened: keyboard.heightWhenOpened.value});
```

Comment:
If you want to keep these logs, we should use the `Log` lib

Reply:
@roryabraham all these logs (as well as commented out code) intended to be removed before a merge. But thanks for the hint - next time I'll use `Log` directly.

Reply:
we should not keep any logs in this code for sure. any animation-related code should be log-free so we don't drop any frames.